### PR TITLE
create exercise 10, fonts and icons

### PR DESCRIPTION
**Pull request by maryx**
Creates exercise 10. Adds custom font and icons.
Files of note:
course/10_icons_fonts/task_10_icons_fonts/pubspec.yaml

course/10_icons_fonts/solution_10_icons_fonts/pubspec.yaml
course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart
course/10_icons_fonts/solution_10_icons_fonts/lib/main.dart
course/10_icons_fonts/solution_10_icons_fonts/lib/category_tile.dart

### CODE REVIEW

File: course/10_icons_fonts/task_10_icons_fonts/pubspec.yaml~ [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: task_10_icons_fonts
```
Comment (yjbanov) [inline course/10_icons_fonts/task_10_icons_fonts/pubspec.yaml~]: You probably don't want to check in this file.

File: course/10_icons_fonts/solution_10_icons_fonts/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: solution_10_icons_fonts
+description: A new Flutter project.
```
Comment (yjbanov) [inline course/10_icons_fonts/solution_10_icons_fonts/pubspec.yaml]: Here and in other pubspecs, maybe we should provide a real description (or remove it if it's optional?).

File: course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    // Dart Usage, see https://www.dartlang.org/guides/language/effective-dart/usage
+    final json = DefaultAssetBundle
+        .of(context)
+        .loadString('assets/data/regular_units.json');
+    final data = JsonDecoder().convert(await json);
+    if (data is! Map) {
+      throw ('Data retrieved from API is not a Map');
+    }
+    var categoryIndex = 0;
+    (data as Map).keys.forEach((key) {
```
Comment (yjbanov) [inline course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart]: same nit

File: course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    setState(() {
+      _currentCategory = category;
+    });
+  }
+
+  /// Makes the correct number of rows for the list view, based on whether the
+  /// device is portrait or landscape.
+  ///
+  /// For portrait, we use a [ListView]
+  /// For landscape, we use a [GridView]
```
Comment (yjbanov) [inline course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart]: same nits

**Comment by yjbanov** [comment]
lgtm

File: course/10_icons_fonts/task_10_icons_fonts/pubspec.yaml~ (hunk 1 above)
Comment (maryx) [reply course/10_icons_fonts/task_10_icons_fonts/pubspec.yaml~]: Ah nice catch!

File: course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart (hunk 3 above)
Comment (maryx) [reply course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart]: Done.

File: course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart (hunk 4 above)
Comment (maryx) [reply course/10_icons_fonts/solution_10_icons_fonts/lib/category_route.dart]: Done.

File: course/10_icons_fonts/solution_10_icons_fonts/pubspec.yaml (hunk 2 above)
Comment (maryx) [reply course/10_icons_fonts/solution_10_icons_fonts/pubspec.yaml]: I'll replace them all with `Work in progress for a Unit Converter App.` Will do this in the PR that changes Future<Null> to Future<void>.

**Pull request merged**